cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -611,7 +621,18 @@ protected void revoke(final EntityId entityId, final Role role, Set<Action> acti
         client.close();
       }
     } catch (Exception e) {
-      throw Throwables.propagate(e);
+      // map sentry exceptions to appropriate cdap-security exceptions
+      if (e instanceof SentryAccessDeniedException || e instanceof SentryGrantDeniedException) {
+        throw Throwables.propagate(new UnauthorizedException(e.getMessage()));
```

Comment:
It would be good if execute can declare the exceptions thrown instead of `Throwables.propagate`